kasa/interfaces/thermostat.py

```diff
@@ -0,0 +1,75 @@
+"""Interact with a TPLink Thermostat."""
+
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from enum import Enum
+from typing import Annotated, Literal
+
+from ..module import FeatureAttribute, Module
+
+
+class ThermostatState(Enum):
+    """Thermostat state."""
+
+    Heating = "heating"
+    Calibrating = "progress_calibration"
+    Idle = "idle"
+    Off = "off"
+    Unknown = "unknown"
+
+
+class Thermostat(Module, ABC):
+    """Base class for TP-Link Thermostat."""
+
+    @property
+    @abstractmethod
+    def state(self) -> bool:
+        """Return thermostat state."""
+
+    @abstractmethod
+    async def set_state(self, enabled: bool) -> dict:
+        """Set thermostat state."""
+
+    @property
+    @abstractmethod
+    def mode(self) -> ThermostatState:
+        """Return thermostat state."""
+
+    @property
+    @abstractmethod
+    def target_temperature(self) -> Annotated[float, FeatureAttribute()]:
+        """Return target temperature."""
+
+    @abstractmethod
+    async def set_target_temperature(
+        self, target: float
+    ) -> Annotated[dict, FeatureAttribute()]:
+        """Set target temperature."""
+
+    @property
+    @abstractmethod
+    def _target_temperature_range(self) -> tuple[int, int]:
+        """Return target temperature range.
+
+        Private method. Consumers of the api should use:
+        get_feature(self.set_target_temperature).minimum_value
+        get_feature(self.set_target_temperature).maximum_value
+        """
```

Comment:
I'm unsure if we want to have privates as part of the interface, but if there's no way around it, I'd stash it to the bottom of the file after methods consumers are expected to use.

Reply:
That's done